test/lib/util/errata_test.rb

```diff
@@ -36,29 +36,60 @@ def setup
       @erratum2.save!
 
       @empty_erratum = Katello::Erratum.create(:errata_id => "empty", :pulp_id => "empty")
+
+      @modular_erratum = ::Katello::Erratum.find_by(errata_id: 'RHEA-2019-002')
+      @modular_rpm = ::Katello::Rpm.create(name: "foobarmod", filename: "foobarmod-1.3.4.rpm", pulp_id: 'foobarmodularrpm')
+      @modular_erratum_package = @modular_erratum.packages.first
+      @module_stream = @modular_erratum_package.module_streams.first
+    end
+
+    # https://projects.theforeman.org/issues/35614
+    def test_filter_by_pulp_href_excludes_errata_with_missing_module_streams
+      # If the erratum's modular RPM count is no longer 1, the test no longer makes sense.
+      # Ensure that all modular packages are accounted for in the final assertion.
+      # The same goes for the number of module streams that the single RPM belongs to.
+      assert_equal 1, @modular_erratum.packages.count
+      assert_equal 1, @modular_erratum_package.module_streams.count
+      assert_equal [@erratum], filter_errata_by_pulp_href([@modular_erratum, @erratum],
+                                                          [@fedora_one.pulp_id, @fedora_two.pulp_id, @modular_rpm.pulp_id],
+                                                          [@fedora_one.filename, @fedora_two.filename, @modular_erratum_package.filename],
+                                                          [@module_stream.module_spec])
+    end
+
+    # https://projects.theforeman.org/issues/35614
+    def test_filter_by_pulp_href_includes_errata_with_proper_module_streams
+      # If the erratum's modular RPM count is no longer 1, the test no longer makes sense.
+      # Ensure that all modular packages are accounted for in the final assertion.
+      # The same goes for the number of module streams that the single RPM belongs to.
+      assert_equal 1, @modular_erratum.packages.count
+      assert_equal 1, @modular_erratum_package.module_streams.count
```

Comment:
these are repeated from the test case above.... can they be extracted from both test cases into a 3rd test case that runs only once?

Reply:
I kept those test cases duplicated because they're testing that the test data is set up correctly.  They're meta-tests in a way. It seems weird to me to have separate unit tests dependent on each other.